src/lib/statistics/statistics_objects/equal_distinct_count_histogram.cpp

```diff
@@ -150,6 +174,399 @@ std::shared_ptr<EqualDistinctCountHistogram<T>> EqualDistinctCountHistogram<T>::
+ * what is required for the current approach.
+ * Some other existing methods such as split_at_bin_bounds also do not work with strings for this reason.
+ * Therefore, string histograms are unmergeable.
+ */
+template <>
+std::tuple<pmr_string, pmr_string, HistogramCountType> EqualDistinctCountHistogram<pmr_string>::_create_one_bin(
+    typename std::vector<pmr_string>::iterator& interval_minima_begin,
+    typename std::vector<pmr_string>::iterator& interval_minima_end,
+    typename std::vector<pmr_string>::iterator& interval_maxima_begin,
+    typename std::vector<pmr_string>::iterator& interval_maxima_end,
+    typename std::vector<HistogramCountType>::iterator& interval_heights_begin,
+    typename std::vector<HistogramCountType>::iterator& interval_heights_end,
+    typename std::vector<HistogramCountType>::iterator& interval_distinct_counts_begin,
+    typename std::vector<HistogramCountType>::iterator& interval_distinct_counts_end, const int distinct_count_target,
+    const HistogramDomain<pmr_string> domain, const bool is_last_bin) {
+  throw std::invalid_argument("Cannot create bin for string histograms.");
+}
+
+template <typename T>
+std::tuple<T, T, HistogramCountType> EqualDistinctCountHistogram<T>::_create_one_bin(
+    typename std::vector<T>::iterator& interval_minima_begin, typename std::vector<T>::iterator& interval_minima_end,
+    typename std::vector<T>::iterator& interval_maxima_begin, typename std::vector<T>::iterator& interval_maxima_end,
+    typename std::vector<HistogramCountType>::iterator& interval_heights_begin,
+    typename std::vector<HistogramCountType>::iterator& interval_heights_end,
+    typename std::vector<HistogramCountType>::iterator& interval_distinct_counts_begin,
+    typename std::vector<HistogramCountType>::iterator& interval_distinct_counts_end, const int distinct_count_target,
+    const HistogramDomain<T> domain, const bool is_last_bin) {
+  const auto remaining_interval_count = std::distance(interval_minima_begin, interval_minima_end);
+  DebugAssert(remaining_interval_count == std::distance(interval_maxima_begin, interval_maxima_end),
+              "All interval vectors need to have the same size.");
+  DebugAssert(remaining_interval_count == std::distance(interval_heights_begin, interval_heights_end),
+              "All interval vectors need to have the same size.");
+  DebugAssert(remaining_interval_count == std::distance(interval_distinct_counts_begin, interval_distinct_counts_end),
+              "All interval vectors need to have the same size.");
+  DebugAssert(remaining_interval_count > 0, "remaining_interval_count has to be greater than zero.");
+
+  const auto bin_start = *interval_minima_begin;
+  auto bin_end = bin_start;
+  auto bin_height = HistogramCountType{0};
+  auto bin_distinct_count = HistogramCountType{0};
+
+  for (auto interval_index = 0u; interval_index < remaining_interval_count; interval_index++,
```

Comment:
auto interval_index = size_t{0};

Reply:
`size_t` is unfortunately not comparable with `std::ptrdiff_t` which we get from `std::distance()`

```
error: comparison of integer expressions of different signedness: ‘long unsigned int’ and ‘const long int’ [-Werror=sign-compare]
  268 |   for (auto interval_index = 0ul; interval_index < remaining_interval_count; interval_index++,
```

Reply:
Then `int64_t`?

Reply:
Now it works.